### DISCUSSION TITLE
Nmea 183 sentence from Engine Hat

### DISCUSSION THREAD

**Discussion by Shelis1976** [hatlabs/discussions#85] [Q&A] [answered]
I have Engine Hat from Hat labs (https://shop.hatlabs.fi/products/sh-esp32-engine-top-hat)
All instaled with LCD, software downloaded, conected to the Signal K serwer on windows.
Data of the RPM, tank A  are displayed on the LCD, and displayed on the section Data Browser on the Signak K server. 
They are dispayed on the Signal K instrument panell too so I think the data comes correct to the Signal K serwer.
I enabled 10110 port on the serwer to put this data to next computer to view them on the Open CPN/QTVLM software.
And No data is transmiting. Connection is ok with the next computer. I/ve checked them putting excample sentence to the data findler. And All is displayed on the Open CPN/QTVLM correctly. 
What should I do to send data from Engine Hat thru Signal K serwer to the Open CPN/QTVLM software?
Signal K to Nmea 183 plugin enable, system K to nmea 2000 enable, N2k to nmea enable....  In the plugin there is no any propulsion.1.revolutions - path .... Should I put them ? where? anable any other plugin?

**Comment by big-col** [comment, marked as answer]
Did you find an answer to this issue?

**Reply by Shelis1976** [reply]
In the plugin there is no  RPM sentence - I made them myself with Node Red. Subscribe puzzle in Node Red in this put the correct path from Signal K, Conect with next puzzle function and put sentence to transleate path to RPM sentense:

var checksum = 0;

sent = "VWRPM,E,0," + msg.payload * 60 + ",50,A"

for (var i = 0; i < sent.length; i++) {
    checksum = checksum ^ sent.charCodeAt(i);
}
checksum = Number(checksum).toString(16)

msg.payload = "$" + sent + "*" + checksum;

return msg;

connect to next puzzle Signal K send nmea 183..

Good luck.

**Comment by mairas** [comment]
Re-opening to provide visibility for future readers.
